go/diff/apply_patch.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package diff
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/types"
+)
+
+// ApplyPatch applies a Patch (list of diffs) to a graph. It makes the following
+// contract:
+//  Given 2 Noms graphs: a1 and a2:
+//    ApplyPatch(a1, Diff(a1, a2)) == a2
+// This is useful for IncrementalUpdate() and possibly other problems. See
+// Updater.go for more information.
+//
+// This function uses a patchStack to maintain state of the graph as it cycles
+// through the diffs in a patch, applying them to 'root' one by one. It minimizes
+// the number of changes to the graph by making subsequent changes locally and
+// incorporating lower level changes into the higher level once all the lower
+// level changes have been made.
+func ApplyPatch(root types.Value, patch Patch) types.Value {
```

Comment:
I guess the reason this comment wasn't clear to me was because I don't understand the last sentence at all. Words like "minimizes" and "subsequent" and "locally" are too abstract ... maybe there is a way to explain the strategy the algorithm uses with more specific words while still staying high enough level to not write a book.

Anyway, it's not a criticism.